web/src/navbar_menus.js

```diff
@@ -13,45 +14,61 @@ export function is_navbar_menus_displayed() {
 }
 
 export function handle_keyboard_events(event_name) {
-    // We don't need to process arrow keys in navbar menus for spectators
-    // since they only have gear menu present.
-    if (
-        popover_menus.is_personal_menu_popover_displayed() &&
-        (event_name === "left_arrow" || event_name === "gear_menu") &&
-        !page_params.is_spectator
-    ) {
-        // Open gear menu popover on left arrow.
-        personal_menu_popover.toggle();
-        gear_menu.toggle();
-        return true;
+    const allowed_events = new Set(["gear_menu", "left_arrow", "right_arrow"]);
+    if (!allowed_events.has(event_name)) {
+        return false;
     }
 
-    if (
-        popover_menus.is_help_menu_popover_displayed() &&
-        (event_name === "right_arrow" || event_name === "gear_menu")
-    ) {
-        // Open gear menu popover on right arrow.
-        navbar_help_menu.toggle();
+    if (event_name === "gear_menu") {
         gear_menu.toggle();
         return true;
     }
+    const $current_navbar_menu = $(".navbar-item:visible").filter(".active");
+    const $next_navbar_menu = get_next_navbar_menu(event_name, $current_navbar_menu);
+
+    if (!$next_navbar_menu) {
+        return false;
+    }
+    change_active_navbar_menu($current_navbar_menu, $next_navbar_menu);
+    return true;
+}
```

Comment:
I'm having a hard time reading what all the things you did here are. It might help to split this first commit into several smaller refactoring commits that are each atomically simple and easy to read.

Check out our [GitHub guide](https://zulip.readthedocs.io/en/latest/git-guide.html) and [commit guidelines](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html) for more details.